### PR TITLE
MRG: Add nirs support to ICA topoplot

### PR DESCRIPTION
**Pull request by rob-luke**
#### Reference issue
Ongoing fNIRS improvements for #7057 

#### What does this implement/fix?
Adds support for plotting Ica components.

### CODE REVIEW

**Comment by agramfort** [comment]
tell us when to look @rob-luke

**Comment by rob-luke** [comment]
Tests look to be passing but codecov has decreased, so trying the test dataset that has overlapping channels to hit those lines.

**Comment by rob-luke** [comment]
@larsoner and @agramfort could you please review this PR

**Comment by rob-luke** [comment]
Sorry I’ve broken the tests. Please don’t review. I will fix tomorrow

**Comment by rob-luke** [comment]
Ok finally all green 😡 

@larsoner and @agramfort could you please review.

**Comment by agramfort** [comment]
rather than branching with an if in each function that calls _merge_grad_data I would suggest to rename _merge_grad_data to _merge_ch_data and do the if branch in a single place. Doing a `git grep _merge_grad_data ` shows you that you made it otherwise do it in other places later

**Comment by rob-luke** [comment]
Thanks @agramfort. Have I understood your suggestion correctly (based on my last commit)? Not ready for another review yet (CI is still yellow), just want to know if I understood your comment correctly.

**Comment by agramfort** [comment]
almost :)

I would rename _merge_grad_data to _merge_ch_data so _merge_ch_data is now
used anywhere you were using so far _merge_grad_data

**Comment by rob-luke** [comment]
Thanks @agramfort. `_merge_ch_data` takes different arguments to `_merge_grad_data` and returns two variables not one. So will take more extensive changes than I can do now. But glad I understand what you mean, I'll get on it.

**Comment by rob-luke** [comment]
@larsoner I can't understand what's causing these CIs to fail. Could you please give me hint.

**Comment by larsoner** [comment]
Travis and Azure are having segfaults in the last few days, unfortunately for now you'll have to look at the logs to make sure your tests are passing :(

**Comment by rob-luke** [comment]
Ok my tests are passing in most CI, except random error. And on my PC.

@agramfort and @larsoner could you please review.

File: mne/evoked.py [hunk 1]
```diff
@@ -312,7 +312,7 @@ def plot_image(self, picks=None, exclude='bads', unit=True, show=True,
     def plot_topo(self, layout=None, layout_scale=0.945, color=None,
                   border='none', ylim=None, scalings=None, title=None,
                   proj=False, vline=[0.0], fig_background=None,
-                  merge_grads=False, legend=True, axes=None,
+                  merge_channels=False, legend=True, axes=None,
```
Comment (agramfort) [inline mne/evoked.py]: you cannot change public APIs like that. We need deprecation cycles.

File: mne/evoked.py [hunk 2]
```diff
@@ -526,7 +526,7 @@ def get_peak(self, ch_type=None, tmin=None, tmax=None,
             Defaults to 'abs'.
         time_as_index : bool
             Whether to return the time index instead of the latency in seconds.
-        merge_grads : bool
+        merge_channels : bool
             If True, compute peak from merged gradiometer data.
```
Comment (agramfort) [inline mne/evoked.py]: same here. And docstrings should then be updated. To be honest I would revert this one unless you need it already for NIRS

File: mne/evoked.py (hunk 1 above)
Comment (agramfort) [reply mne/evoked.py]: To be honest I would revert this one unless you need it already for NIRS

Comment (rob-luke) [reply mne/evoked.py]: No don't need this at all. I just misunderstood a previous comment and thought I was meant to change all occurrences of merge grad to merge channels. I'll change this back

**Comment by agramfort** [comment]
sorry I meant internal calls not public API

>

**Comment by rob-luke** [comment]
all good. I will leave all internal calls at merge_channels, and revert all public api to merge_grads.

File: mne/viz/topo.py [hunk 3]
```diff
@@ -568,7 +568,7 @@ def _plot_evoked_topo(evoked, layout=None, layout_scale=0.945, color=None,
                       border='none', ylim=None, scalings=None, title=None,
                       proj=False, vline=(0.,), hline=(0.,), fig_facecolor='k',
                       fig_background=None, axis_facecolor='k', font_color='w',
-                      merge_grads=False, legend=True, axes=None, show=True,
+                      merge_channels=False, legend=True, axes=None, show=True,
```
Comment (rob-luke) [inline mne/viz/topo.py]: Is this considered public API? Should I refer this change too?

**Comment by agramfort** [comment]
what is public API is any function or class that does **not** start with _

so here you're good.

File: mne/viz/topomap.py [hunk 4]
```diff
@@ -334,14 +334,15 @@ def plot_projs_topomap(projs, info, cmap=None, sensors=True,
         data = proj['data']['data'].ravel()
         info_names = _clean_names(info['ch_names'], remove_whitespace=True)
         use_info = pick_info(info, pick_channels(info_names, ch_names))
-        data_picks, pos, merge_grads, names, _, this_sphere, clip_origin = \
+        data_picks, pos, merge_channels, names, _, this_sphere, clip_origin = \
```
Comment (agramfort) [inline mne/viz/topomap.py]: revert this is public

Comment (agramfort) [reply mne/viz/topomap.py]: my bad

**Comment by larsoner** [comment]
Thanks @rob-luke

**Pull request merged**